### ISSUE TITLE
Preprocess single barcoded reads

### ISSUE THREAD

**Issue by kweitemier** [msettles/dbcAmplicons#41]
Hello,

I notice that the update history for version 0.9.0 states that single barcoded reads can be processed, but I can't figure out how to do that.

Is there an option I need to set or a way to set up the input files to do this?

I have paired-end reads with a third read containing the barcodes.
For the Barcode file I put just two columns (BarcodeID, BarcodeSeq).
I then ran preprocess with `--R1 Seq_R1.fastq --R2 Seq_R2.fastq --BC1 Barcodes.fastq`

I get the following Traceback, which I interpret as program missing the other barcode read:
```
  File "build/bdist.linux-x86_64/egg/dbcAmplicons/preprocess_app.py", line 161, in start
    self.run = FourReadIlluminaRun(fastq_file1, fastq_file2, fastq_file3, fastq_file4)
  File "build/bdist.linux-x86_64/egg/dbcAmplicons/illuminaRun.py", line 48, in __init__
    self.fbc2.append(misc.infer_read_file_name(fread, "3"))
  File "build/bdist.linux-x86_64/egg/dbcAmplicons/misc.py", line 104, in infer_read_file_name
    raise Exception("Error inferring read " + seakread + " from read 1, found " + str(len(read)) + " suitable matches.")
Exception: Error inferring read 3 from read 1, found 0 suitable matches.
```

Thanks for the help!

**Comment by pmhenry** [comment]
Apologies for the delayed response. In case this is still relevant, it looks like you are trying to put your barcode lookup table in fasta format, but the program is designed for the barcode table to be in a tab delimited format. See the examples in the "BarcodePrimersTable" folder and try to replicate. Feel free to respond if you have further questions. I'll get to it quicker next time.

**Comment by kweitemier** [comment]
Thanks for the response.
I've found a workaround, but never solved the original problem -- although maybe I misunderstood what functionality was intended.
The sequences I'm working with come as three fastq files: forward read 1, reverse read 2, and a barcoding read (usually 8 or 10 bp).
I took the new function described in version 0.9.0, "single barcoded reads can be processed", to mean that dbcAmplicons could handle this arrangement. But looking now, maybe that only refers to reads with inline barcodes.

My workaround was to create a dummy fastq file to be the fourth read, barcoding read 2. I then could create the Barcode file with the proper three columns and run the command with `dbcAmplicons preprocess -O outdir -B BarcodeFile_with_dummy -P PrimerFile -S MetaFile --R1 seqs_R1.fastq --R2 seqs_R2.fastq --BC1 seqs_Index1.fastq --BC2 Dummy_Index2.fastq`.
